components/api/src/main/java/com/hotels/styx/api/LiveHttpRequest.java

```diff
@@ -383,6 +383,11 @@ public Transformer newBuilder() {
                 .findFirst();
     }
 
+    @Override
+    public Eventual<LiveHttpRequest> consume() {
+        return MessageBodyConsumption.consume(this, LiveHttpRequest.class);
```

Comment:
HTTP message body is the same for both requests and responses. Therefore you shouldn't have to pass in a class as a  parameter.